### PR TITLE
COMPAT: add back block manager constructors to pandas.core.internals API

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Those were removed in a for the rest unrelated PR (https://github.com/pandas-dev/pandas/pull/33100, cc @jbrockmendel)

But at least `create_block_manager_from_blocks` is used by dask (in partd: https://github.com/dask/partd/blob/master/partd/pandas.py), so dask's tests are failing with pandas master (cc @TomAugspurger dask isn't testing with pandas master?). But so to be safe just added back both.

### CODE REVIEW

**Comment by TomAugspurger** [comment]
Dask does test against pandas master, but it's allowed to fail. Looking
into it this week.

**Comment by jbrockmendel** [comment]
Is weaning downstream packages off of these a viable option?

File: pandas/core/internals/__init__.py [hunk 1]
```diff
@@ -33,4 +38,6 @@
     "BlockManager",
     "SingleBlockManager",
     "concatenate_block_managers",
+    "create_block_manager_from_arrays",
+    "create_block_manager_from_blocks",
```
Comment (jbrockmendel) [inline pandas/core/internals/__init__.py:42]: can you add a comment about why these are needed here

**Comment by jorisvandenbossche** [comment]
> Is weaning downstream packages off of these a viable option?

That is certainly something to look at, but apart from that, we still need to add it back for compat with released versions IMO (and we could also deprecate it first before removing again)

**Comment by jbrockmendel** [comment]
CI fails unrelated; LGTM

**Comment by TomAugspurger** [comment]
It'd be a bit hard for `partd` to move off this. The purpose is to (de)serialize a DataFrame as quickly as possible, which essentially requires working at the block level.

**Comment by jorisvandenbossche** [comment]
> It'd be a bit hard for partd to move off this. The purpose is to (de)serialize a DataFrame as quickly as possible, which essentially requires working at the block level.

Yes, that's exactly the same reason that pyarrow also interacts with the BlockManager. 

Looking at pyarrow, there we are using the main constructor: `BlockManager(reconstructed_blocks, axes)`, which seems basically what `create_block_manager_from_blocks` does (the function additionally calls `_consolidate_inplace()`). 
But, I am personally not sure it is necessarily "better" to use `BlockManager(..)` over `create_block_manager_from_blocks(..)`. Although in theory it would be nice that all external projects use the same method to create BlockManager.

**Comment by TomAugspurger** [comment]
If we want, we could create a restricted "publich-ish but not necessarily stable" dev API for projects to interact with the block manager. That way we at least know what projects are using.

Dunno if that's worth the effort though. The current setup of downstream projects testing against master seems to work out OK.

**Comment by jorisvandenbossche** [comment]
For me that "dev API" to interact with blocks is basically what is exposed in `pandas.core.internals` now? (maybe with the exception of `concatenate_block_managers`)

**Comment by TomAugspurger** [comment]
Haven't really looked, but that sounds fine :)

**Comment by jbrockmendel** [comment]
@jorisvandenbossche before i make a PR removing _safe_reshape from `internals.__init__`, are you aware of any downstream packages that need it there?

**Comment by jorisvandenbossche** [comment]
I am not aware of any `_`-prefixed functions being used

**Comment by jbrockmendel** [comment]
@phofl does dask still need create_block_manager_from_blocks?

**Comment by phofl** [comment]
yeah that is still in use

**Pull request merged**